### PR TITLE
Fix Rotation Bug

### PR DESCRIPTION
**Pull request by evancrow**
Needed to pass GeomReader down from outside the `NavigationView`.

### How was this tested?
- [x] I tested this manually
- [ ] I added automated tests
- [ ] Existing automated tests are enough to cover my changes

### Screenshots and screen recordings
https://user-images.githubusercontent.com/37356984/158844851-8b48ca21-e861-4668-997e-e24edd4fa3c0.mp4

## Issues addressed
Fixed: #3143

### CODE REVIEW

**Comment by chung-lai** [comment]
Build test is failing of missing `geom` parameter.

**Review by chung-lai — COMMENTED** [review]
The change looks good to me. Is that navigation view coexist with BrowserView so we need to pass down the geometry reader? And now TabSwitcher is a navigationView?

Would love a pass on this change from @yozuysal who reviewed the original navigationview code.

**Comment by evancrow** [comment]
That's basically it, and reading the Geom inside the `NavigationView` broke the `CardGrid` on rotation.

File: Client/Frontend/Browser/Card/CardGrid.swift [hunk 1]
```diff
@@ -141,39 +149,37 @@ struct CardGrid: View {
...
-                NavigationLink(
-                    destination: WalletDetailView()
-                        .environment(\.cardSize, cardSize)
-                        .environment(\.aspectRatio, CardUX.DefaultTabCardRatio),
-                    isActive: $walletDetailsModel.showingWalletDetails
-                ) {}
+            NavigationLink(
+                destination: detailedSpaceView,
+                isActive: $gridModel.showingDetailView
+            ) {}.useEffect(deps: spaceModel.detailedSpace) { detailedSpace in
```
Comment (yozuysal) [inline Client/Frontend/Browser/Card/CardGrid.swift:169]: so we dont need to updateCardSize here anymore? Just double checking we carried over the right logic for useEffect

File: Client/Frontend/Browser/Card/CardsContainer.swift [hunk 2]
```diff
@@ -117,78 +118,79 @@ struct CardsContainer: View {
...
+            // Spaces
+            CardScrollContainer(columns: columns) { scrollProxy in
+                VStack(alignment: .leading) {
+                    LazyVGrid(columns: columns, spacing: CardGridUX.GridSpacing) {
+                        SpaceCardsView()
+                            .environment(\.columns, columns)
+                    }.animation(nil)
                 }
-                .offset(x: (gridModel.switcherState == .spaces ? 0 : geom.size.width))
-                .accessibilityElement(children: .contain)
```
Comment (yozuysal) [inline Client/Frontend/Browser/Card/CardsContainer.swift:147]: I am not sure if we need these accessibility modifiers moving in this PR tbh. It is already a bit involved as is.

**Review by yozuysal — APPROVED** [review]
the code looks OK to me, but honestly it is quite a big shift for a change we are hoping to cherry-pick on the release day. I feel like without a whole lot of test driving this, we will definitely miss some regression in the larger grid layout...

**Comment by chung-lai** [comment]
> @darinwf @chassu `tabCard.manager.get(for: tabCard.id)!` is crashing for `ClearPrivateDataTests`

hmm..looks flaky to me as it didn't crash in the previous build? Should we try rerunning it?

**Comment by chung-lai** [comment]
Went ahead and rerunning ui-tests. Here is the log for the crashed test if we want to look: https://app.circleci.com/pipelines/github/neevaco/neeva-ios/1192/workflows/12a91810-9133-4f63-871d-f6b45e75c38c/jobs/5536

**Review by yozuysal — APPROVED** [review]

**Review by chung-lai — APPROVED** [review]

**Comment by chassu** [comment]
I thought an earlier fix made sure allDetails and tabManager.tabs are always in sync... Will keep an eye on the rerun

**Comment by evancrow** [comment]
Okay, everything passed, going to merge.

File: Client/Frontend/Browser/BrowserView.swift [hunk 3]
```diff
@@ -101,102 +101,105 @@ struct BrowserView: View {
...
+            ZStack {
+                NavigationView {
+                    mainContent
+                        .navigationBarHidden(true)
+                }.navigationViewStyle(.stack)
+
+                OverlayView(overlayManager: overlayManager)
+            }
+            .useEffect(deps: geom.size) { _ in
+                self.geom = geom
```
Comment (darinwf) [inline Client/Frontend/Browser/BrowserView.swift:185]: Oh, perhaps the bug here is that we only have a `deps` on geom.size and not geom.safeAreaInsets?

**Pull request merged**